vspec/model/vsstree.py

```diff
@@ -149,61 +203,84 @@ def extractCoreAttribute(name: str):
                 self.data_type_str = self.source_dict["datatype"]
                 self.validate_and_set_datatype()
             else:
-                logging.error("Item %s cannot have datatype, only allowed for signal and property!", self.name)
+                logging.error(
+                    "Item %s cannot have datatype, only allowed for signal and property!",
+                    self.name,
+                )
                 sys.exit(-1)
-        elif (self.is_signal() or self.is_property()):
+        elif self.is_signal() or self.is_property():
             raise IncompleteElementException(
-                (f"Incomplete element {self.name} from {self.source_dict['$file_name$']}: "
-                 f"Elements of type {self.type.value} need to have a datatype declared."))
+                f"Incomplete element {self.name} from {self.source_dict['$file_name$']}: "
+                f"Elements of type {self.type.value} need to have a datatype declared."
+            )
 
         # Datatype check for unit performed first when we have set the right datatype
         if self.has_unit():
 
             if not self.has_datatype():
-                logging.error("Unit specified for item not using standard datatype: %s", self.name)
+                logging.error(
+                    "Unit specified for item not using standard datatype: %s", self.name
+                )
                 sys.exit(-1)
 
         if self.has_instances() and not self.is_branch():
             logging.error(
-                f"Only branches can be instantiated. {self.qualified_name()} is of type {self.type}")
+                f"Only branches can be instantiated. {self.qualified_name()} is of type {self.type}"
+            )
             sys.exit(-1)
 
+        if self.has_state():
+            self.state = VSSNodeState(self.source_dict["state"])
+            if self.state == VSSNodeState.DELETED:
+                self.parent = None
+                self.children = []
+
```

Comment:
for anyone reading this, i formatted the whole file because it was hurting my eyes, this is the change you might be looking for